server/auth/user.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissionKeys and
+// limitations under the License.
+
+package auth
+
+import (
+	"encoding/json"
+	"sort"
+)
+
+// User records user info.
+// Read-Only once created.
+type User struct {
+	username string
+	hash     string
+	roleKeys map[string]struct{}
+}
+
+type jsonUser struct {
+	Username string   `json:"username"`
+	Hash     string   `json:"hash"`
+	RoleKeys []string `json:"roles"`
+}
```

Comment:
We can maintain the model in a certain file, and directly define the json tag in `User`